### PR TITLE
Add ability to configure HTTP Basic auth for Nginx

### PR DESCRIPTION
**Pull request by dznz**
Override the default `nginx` recipes to allow for http basic auth.

/cc @charlespeach

### CODE REVIEW

File: nginx/attributes/nginx.rb [hunk 1]
```diff
@@ -37,6 +37,8 @@
 
 default[:nginx][:log_format] = {}
 
+default[:nginx][:auth_basic_user_file] = "conf.d/htpasswd"
```
Comment (charlespeach) [inline nginx/attributes/nginx.rb:40]: should this directory be relative or a full path? I only ask because I have no idea! :+1:

**Comment by charlespeach** [comment]
Nice @dznz just had the one question above. No idea how you would test this

File: nginx/attributes/nginx.rb (hunk 1 above)
Comment (dznz) [inline nginx/attributes/nginx.rb:40]: I believe it's relative, per the [docs example](http://nginx.org/en/docs/http/ngx_http_auth_basic_module.html) but who knows. I'll fix up if it doesn't work :grin:

**Comment by jackcallister** [comment]
We test it by firing from the hip.

**Comment by dznz** [comment]
Shit yeah we do!

**Pull request merged**